### PR TITLE
CLD-1037-deployments-v1-errors

### PR DESCRIPTION
**Pull request by SohailHaqyar**

### CODE REVIEW

File: src/http-exception.ts [hunk 1]
```diff
@@ -4,29 +4,33 @@ import {
...
-  generateErrorOriginCode,
+  routeMap,
+  Details,
 } from '.';
 
 class HttpException {
   code: number;
   status: HttpStatusCode;
   reason: ErrorReason | string;
+  details?: string;
```
Comment (TonyH-Salient) [inline src/http-exception.ts:15]: Is this param still needed?

File: src/error-details/users.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { Details, IDetailsMap } from '..';
+
+export const usersErrorDetailsMap: IDetailsMap = {
+  [Details.UserIdMissingFromLocals]: {
+    friendlyMessage: 'Bad token',
```
Comment (TonyH-Salient) [inline src/error-details/users.ts:5]: This doesn't seem right

File: src/error-details/shared.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+  [Details.JwtVerifyFailed]: {
+    friendlyMessage: 'Bad token',
+    technichalReason: 'jwt.verify failed to verify the token, and errored out',
+  },
+
+  [Details.NoTokenAudience]: {
+    friendlyMessage: 'Bad token',
+    technichalReason: '(in jwt.verify) The token.aud returned a falsy value',
+  },
```
Comment (TonyH-Salient) [inline src/error-details/shared.ts:24]: These 4 can be moved into a separate file, perhaps `auth.ts`?

File: src/enums/details.enum.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  'OrganizationByUserIdNotFoundInDb',
+  'OrganizationTypeNotSalient',
+  'OrganizationTypeNotIntegrator',
+  'OrganizationTypeNotEndUser',
+  'OrganizationMissingFromLocals',
+  'OrganizationByDeploymentIdNotFoundInDb',
+  'OrgNotFoundWhenDeregisteringDeployment',
+] as const;
+
+const Deployments = [
```
Comment (TonyH-Salient) [inline src/enums/details.enum.ts:41]: Does it make sense to move constants in this file into their corresponding file? E.g: `Organization` into `error-details/organization.ts`

**Review by TonyH-Salient — CHANGES_REQUESTED** [review]

File: src/enums/details.enum.ts (hunk 4 above)
Comment (SohailHaqyar) [reply src/enums/details.enum.ts]: yup: [1fbec31](https://github.com/SalientSys/salient-sst-errors/pull/8/commits/1fbec3138ca51e902fae455754c04838e481dd7f)

File: src/error-details/shared.ts (hunk 3 above)
Comment (SohailHaqyar) [reply src/error-details/shared.ts:24]: [1fbec31](https://github.com/SalientSys/salient-sst-errors/pull/8/commits/1fbec3138ca51e902fae455754c04838e481dd7f)

File: src/error-details/users.ts (hunk 2 above)
Comment (SohailHaqyar) [reply src/error-details/users.ts]: co pilot [1fbec31](https://github.com/SalientSys/salient-sst-errors/pull/8/commits/1fbec3138ca51e902fae455754c04838e481dd7f)

File: src/http-exception.ts (hunk 1 above)
Comment (SohailHaqyar) [reply src/http-exception.ts:13]: Nope [1fbec31](https://github.com/SalientSys/salient-sst-errors/pull/8/commits/1fbec3138ca51e902fae455754c04838e481dd7f)

**Pull request merged**